model/src/main/kotlin/ResolvedConfiguration.kt

```diff
@@ -28,10 +33,71 @@ import com.fasterxml.jackson.annotation.JsonInclude
  * TODO: Add further data.
  */
 data class ResolvedConfiguration(
+    @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = PackageCurationsFilter::class)
+    val packageCurations: PackageCurations = PackageCurations(),
+) {
+    /**
+     * Return all [PackageCuration]s contained in this [ResolvedConfiguration] in highest-priority-first order.
+     */
+    @JsonIgnore
+    fun getAllPackageCurations(): List<PackageCuration> =
+        packageCurations.providers.flatMap { id ->
+            packageCurations.data[id].orEmpty()
+        }
+}
+
+@JsonSerialize(converter = PackageCurationsConverter::class)
+data class PackageCurations(
     /**
-     * All package curations applicable to the packages contained in the enclosing [OrtResult] ordered
-     * highest-priority-first.
+     * All enabled providers ordered highest-priority-first.
      */
-    @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val packageCurations: List<PackageCuration> = emptyList()
-)
+    val providers: List<String> = emptyList(),
+
+    /**
+     * All package curations applicable to the packages contained in the enclosing [OrtResult].
+     */
+    val data: Map<String, List<PackageCuration>> = emptyMap()
+) {
+    @JsonIgnore
+    fun isEmpty(): Boolean =
+        providers.isEmpty() && data.isEmpty()
+
+    init {
+        val duplicateProviders = providers.getDuplicates()
+        require(duplicateProviders.isEmpty()) {
+            "The list 'providers' contains the following duplicates, which is not allowed: " +
+                    "${duplicateProviders.joinToStringSingleQuoted()}."
+        }
+
+        val invalidProviderReferences = data.keys.filter { it !in providers }
+        require(invalidProviderReferences.isEmpty()) {
+            "The following keys in 'data' reference a non-existing provider, which is not allowed: " +
+                    "${invalidProviderReferences.joinToStringSingleQuoted()}."
+        }
+    }
+}
+
+@Suppress("EqualsOrHashCode", "EqualsWithHashCodeExist") // The class is not supposed to be used with hashing.
+private class PackageCurationsFilter {
+    override fun equals(other: Any?): Boolean =
+        other is PackageCurations && other.isEmpty()
+}
+
+/**
+ * Sorts the keys of [PackageCurations.data] alphabetically and the values of [PackageCurations.data] by
+ * their [id][PackageCuration.id]. Drops entries for provider without curations and removes duplicate curations for
```

Comment:
Why is this still necessary with all the `distinct()` calls we now have in the provider implementations? Shouldn't we only have either-or?

Reply:
I decided for the distinct calls because IMO it just encodes the appropriate way of serializing an instance of this class. I believe it's a good thing to not introduce such assumption, considering also programatical use or (future) use from code paths we currently don't have in mind.

Does it make sense? (If not I'll ditch my preference for you)

Reply:
IMO we should have `distinct()` only in one place, and preferably in a place what does not require / rely on all curation provider implementers to remember calling `distinct()`. So I guess we should have it here, but not in the implementations. Or extend the `PackageCurationProvider` with a public and an internal function, where the public one just calls the internal one plus `distinct()`, and implementers only implement the internal function.

Reply:
From my perspective the `distinct()` call should be present in both, (1) the providers and (2) the `ResolvedConfiguration` serialization. I argue that both components can be used independently from one another and considering both in isolation it makes sense to me that 1. providers return distinct configurations and 2. serialization drops duplicates. I'm also fine with just disagreeing on that one.

I'm not sure what exactly you have in mind with "public and an internal function", but I have a hunch that this could again be out of scope / separate commit if at all.

Reply:
I still fear this is some kind of mutual misunderstanding involved in the discussion, maybe let's have a call to shortcut this.

Reply:
I agree having a call would help. Maybe we could even shift the developer meeting from next Monday to tomorrow already, as on Monday I'll probably not be available.

Reply:
That'd be awesome. I'll propose it tomorrow morning in the group chat.

Reply:
The distinct calls have been eliminated as discussed:
- the provider API now returns sets of curations, done in a separate already merged PR.
- the `ResolvedConfiguration` now also holds sets instead of lists of curations.